state-chain/runtime/src/chainflip.rs

```diff
@@ -1021,3 +1021,49 @@ impl cf_traits::MinimumDeposit for MinimumDepositProvider {
 		}
 	}
 }
+
+pub struct CfTransactionFeeScaler;
+impl TransactionFeeScaler<RuntimeCall, AccountId, FlipBalance> for CfTransactionFeeScaler {
+	fn call_info_and_spam_prevention_upfront_fee(
+		call: &RuntimeCall,
+		caller: &AccountId,
+	) -> Option<(CallInfoId<AccountId>, FlipBalance)> {
+		match call {
+			RuntimeCall::LiquidityPools(pallet_cf_pools::Call::set_limit_order {
+				base_asset,
+				..
+			}) |
+			RuntimeCall::LiquidityPools(pallet_cf_pools::Call::update_limit_order {
+				base_asset,
+				..
+			}) |
+			RuntimeCall::LiquidityPools(pallet_cf_pools::Call::set_range_order {
+				base_asset,
+				..
+			}) |
+			RuntimeCall::LiquidityPools(pallet_cf_pools::Call::update_range_order {
+				base_asset,
+				..
+			}) => Some(CallInfoId::Pool(PoolTouched {
+				account: caller.clone(),
+				base_asset: *base_asset,
+			})),
+			_ => None,
+		}
+		.map(|call_info_id| (call_info_id, FLIPPERINOS_PER_FLIP))
+	}
+
+	fn scale_fee(
+		fee_scaling_config: FeeScalingRateConfig,
+		pre_scaled_fee: FlipBalance,
+		call_count: u16,
+	) -> FlipBalance {
+		match fee_scaling_config {
+			FeeScalingRateConfig::ExponentBuffer { buffer, exp_base } => {
+				let multiplier = exp_base.saturating_pow(call_count.saturating_sub(buffer).into());
+				multiplier.saturating_mul_int(pre_scaled_fee)
+			},
+			FeeScalingRateConfig::NoScaling => pre_scaled_fee,
+		}
```

Comment:
I think this is wrong. We want to scale the fee *down*.

On the first (or zeroth) call, the exponent is always zero, so the multiplier will always be one. 

Also, it feels a bit weird having the fee scaling implemented in this trait, IMO would be more natural to have the impl co-located with the enum definition and then just call a method here: 

```rust
fn scale_fee(
		fee_scaling_config: FeeScalingRateConfig,
		pre_scaled_fee: FlipBalance,
		call_count: u16,
	) -> FlipBalance {
		fee_scaling_config.scale(pre_scaled_fee, call_count)
	}
```

Reply:
the tests seem to work as expected, not sure what you mean

Reply:
I mean that we deduct 1 FLIP before execution, we can't burn more than that, we can only *reduce* the fee, we can't increase it. This function takes the already deducted fee and then scales it up - or am I misreading it? 

But we can't burn more than we have already deducted. For one, we can't guarantee that the account has sufficient balance.